src/MongoDB.EntityFrameworkCore/Serializers/EntitySerializer.cs

```diff
@@ -48,7 +53,49 @@ public TValue Deserialize(BsonDeserializationContext context, BsonDeserializatio
             => Deserialize(context, args);
 
         public void Serialize(BsonSerializationContext context, BsonSerializationArgs args, TValue value)
-            => throw new NotImplementedException();
+        {
+            if (value == null)
+            {
+                context.Writer.WriteNull();
+                return;
+            }
+
+            context.Writer.WriteStartDocument();
+            WriteKey(context.Writer, value);
```

Comment:
Does every class have a key?

Do we not support compound types that don't have a key, like address?

Reply:
So this comment led to a thought which led to a discussion with the EF team on identity and what it means.

The preference from the EF world is that root entities have identity and thus should be compared only by identity for a variety of reasons.

Meanwhile entities without keys (such as owned entities by default) have no identity (key) and so should be compared by value.

I think we need to dig more into the comparison of key based entities as if we go down the EF route we'll have to rewrite the tree to change entity comparisons to be by key rather than serializing the entity. If we decide not to go down that route but to align more with the existing C# LINQ driver then we'll need to understand and document the implications.

In the mean-time we'll throw for that case so we can proceed with null comparisons (the original issue) and keyless by-value comparisons (no debate).